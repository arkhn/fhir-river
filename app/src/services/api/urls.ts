let API_URL = "" as string | undefined;
if (process.env.NODE_ENV === "production") {
  ({ REACT_APP_API_URL: API_URL } = (window as any).env);
} else {
  ({ REACT_APP_API_URL: API_URL } = process.env);
}

export { API_URL };
