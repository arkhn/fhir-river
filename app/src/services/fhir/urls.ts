export const { REACT_APP_FHIR_API_URL: FHIR_API_URL } = process.env;
