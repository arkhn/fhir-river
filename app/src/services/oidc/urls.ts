export const {
  REACT_APP_OIDC_LOGIN_URL: OIDC_LOGIN_URL,
  REACT_APP_OIDC_LOGOUT_URL: OIDC_LOGOUT_URL,
} = process.env;
