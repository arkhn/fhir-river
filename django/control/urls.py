from django.urls import include, path

from control.api import views
from control.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
    path("delete-resources/", views.ResourceEndpoint.as_view(), name="delete-resources"),
]
