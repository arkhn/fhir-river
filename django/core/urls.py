from django.urls import include, path

from core.api import views
from core.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("version/", views.VersionEndpoint.as_view()),
]
