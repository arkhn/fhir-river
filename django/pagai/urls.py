from django.urls import path

from pagai.api import views

urlpatterns = [
    path("pagai/list-owners/", views.OwnersListView.as_view(), name="list-owners"),
    path("pagai/owner-schema/<owner>/", views.OwnerSchemaView.as_view(), name="owner-schema"),
    path("pagai/explore/<resource_id>/<owner>/<table>/", views.ExploreView.as_view(), name="explore"),
]
