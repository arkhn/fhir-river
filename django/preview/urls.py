from django.urls import include, path

from preview.api.router import router
from preview.api import views

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
]
