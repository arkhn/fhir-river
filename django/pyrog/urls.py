from django.urls import include, path

from pyrog.api import views
from pyrog.api.router import router


urlpatterns = [
    path("api/", include(router.urls)),
    path('session/', views.Session.as_view()),
    path('oidc/', include('mozilla_django_oidc.urls')),
]
