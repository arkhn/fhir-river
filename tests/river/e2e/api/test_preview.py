from uuid import uuid4

import pytest

from django.urls import reverse

pytestmark = pytest.mark.django_db


def test_preview(api_client, users_to_patients_mapping):
    url = reverse("preview")

    data = {
        "mapping": users_to_patients_mapping,
        "primary_key_values": [uuid4()],
    }
    response = api_client.post(url, data, format="json")

    assert response.status_code == 200
