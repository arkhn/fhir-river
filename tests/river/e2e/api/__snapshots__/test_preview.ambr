# name: TestPreview.test_preview
  <class 'dict'> {
    'birthDate': '2094-03-05',
    'deceasedDateTime': '2165-08-12',
    'gender': 'female',
    'id': '6140cc54-cd63-5ad9-b3f9-000649e5368f',
    'identifier': <class 'list'> [
      <class 'dict'> {
        'value': '9467 (1)',
      },
    ],
    'maritalStatus': <class 'dict'> {
      'coding': <class 'list'> [
        <class 'dict'> {
          'code': 'SEPARATED',
        },
      ],
    },
    'meta': <class 'dict'> {
      'tag': <class 'list'> [
        <class 'dict'> {
          'system': 'http://terminology.arkhn.org/CodeSystem/source',
        },
        <class 'dict'> {
          'system': 'http://terminology.arkhn.org/CodeSystem/resource',
        },
      ],
    },
    'resourceType': 'Patient',
  }
---
# name: TestPreview.test_preview_with_validation_error
  <class 'dict'> {
    'code': 'invalid',
    'diagnostics': 'shit happened',
    'severity': 'error',
  }
---
# name: TestPreview.test_preview_with_validation_exception
  <class 'dict'> {
    'code': '',
    'diagnostics': 'fhir-api-returned-an-internal-server-error',
    'severity': 'error',
  }
---
